### PR TITLE
Documentation update - daal4py, model builders 

### PR DESCRIPTION
**Pull request by napetrov**
# Description
Documentation update and some cleanup 

Changes proposed in this pull request:
- Refreshing daal4py documentation
- Adding first version of Model Builders docs
- Removing empty examples

### CODE REVIEW

File: doc/daal4py/index.rst [hunk 1]
```diff
@@ -313,6 +189,9 @@ daal4py will work on the entire file content.
 
 Scikit-Learn API and patching
 -----------------------------
+.. note::
+    Recommended flow for scikit-lear patching is through use of the
```
Comment (Alexsandruss) [inline doc/daal4py/index.rst:193]:
```suggestion
    Recommended flow for scikit-learn patching is through use of the 
```

File: doc/daal4py/examples.rst [hunk 2]
```diff
@@ -20,15 +20,15 @@ Examples
...
-Data Science examples
----------------------
+General usage
+-------------
 
-Jupyter Notebooks
+Model Builders models conversion
 
-- `Linear Regression <https://github.com/intel/scikit-learn-intelex/tree/master/examples/daal4py/daal4py_data_science.ipynb>`_
+- `XGBoost model conversion <https://github.com/intel/scikit-learn-intelex/blob/master/examples/daal4py/gbt_cls_model_create_from_xgboost_batch.py>`_
```
Comment (alexandraepan) [inline doc/daal4py/examples.rst:28]:
```suggestion
- `XGBoost* model conversion <https://github.com/intel/scikit-learn-intelex/blob/master/examples/daal4py/gbt_cls_model_create_from_xgboost_batch.py>`_
```

File: doc/daal4py/examples.rst [hunk 3]
```diff
@@ -20,15 +20,15 @@ Examples
...
+- `LightGBM model conversion <https://github.com/intel/scikit-learn-intelex/blob/master/examples/daal4py/gbt_cls_model_create_from_lightgbm_batch.py>`_
+- `CatBoost model conversion <https://github.com/intel/scikit-learn-intelex/blob/master/examples/daal4py/gbt_cls_model_create_from_catboost_batch.py>`_
```
Comment (alexandraepan) [inline doc/daal4py/examples.rst:30]:
```suggestion
- `CatBoost* model conversion <https://github.com/intel/scikit-learn-intelex/blob/master/examples/daal4py/gbt_cls_model_create_from_catboost_batch.py>`_
```

File: doc/daal4py/index.rst [hunk 4]
```diff
@@ -99,39 +99,8 @@ Last but not least, daal4py allows :ref:`getting input data from streams <stream
...
-DPC++ compiler runtime can be installed either from PyPI or Anaconda:
-
-- Install from PyPI::
-
-     pip install dpcpp-cpp-rt
-
-- Install from Anaconda::
-
-     conda install dpcpp_cpp_rt -c intel
+daal4py oneAPI and GPU support deprecated - please use scikit-learn-intelex
```
Comment (alexandraepan) [inline doc/daal4py/index.rst:102]:
```suggestion
daal4py oneAPI and GPU support is deprecated. Use `scikit-learn-intelex <https://intel.github.io/scikit-learn-intelex/oneapi-gpu.html#>`_ 
```

File: doc/daal4py/index.rst [hunk 5]
```diff
@@ -99,39 +99,8 @@ Last but not least, daal4py allows :ref:`getting input data from streams <stream
...
-
-- Install from PyPI::
-
-     pip install dpcpp-cpp-rt
-
-- Install from Anaconda::
-
-     conda install dpcpp_cpp_rt -c intel
+daal4py oneAPI and GPU support deprecated - please use scikit-learn-intelex 
+instead -  https://intel.github.io/scikit-learn-intelex/oneapi-gpu.html#
```
Comment (alexandraepan) [inline doc/daal4py/index.rst:103]:
```suggestion
instead.
```

File: doc/daal4py/index.rst [hunk 6]
```diff
@@ -174,7 +143,11 @@ daal4py is available at the `Python Package Index <https://pypi.org/project/daal
 on Anaconda Cloud in `Conda Forge channel <https://anaconda.org/conda-forge/daal4py>`_
 and in `Intel channel <https://anaconda.org/intel/daal4py>`_.
 Sources and build instructions are available in
-`daal4py repository <https://github.com/IntelPython/daal4py>`_.
+`daal4py repository <https://github.com/intel/scikit-learn-intelex/tree/master/daal4py>`_.
+
+daal4py package avaialbe on same channels and platforms as scikit-learn-intelex, please
```
Comment (alexandraepan) [inline doc/daal4py/index.rst:148]:
```suggestion
The daal4py package is available via same distribution channels and platforms as scikit-learn-intelex. 
```

File: doc/daal4py/index.rst [hunk 7]
```diff
@@ -174,7 +143,11 @@ daal4py is available at the `Python Package Index <https://pypi.org/project/daal
 on Anaconda Cloud in `Conda Forge channel <https://anaconda.org/conda-forge/daal4py>`_
 and in `Intel channel <https://anaconda.org/intel/daal4py>`_.
 Sources and build instructions are available in
-`daal4py repository <https://github.com/IntelPython/daal4py>`_.
+`daal4py repository <https://github.com/intel/scikit-learn-intelex/tree/master/daal4py>`_.
+
+daal4py package avaialbe on same channels and platforms as scikit-learn-intelex, please 
+refer to
```
Comment (alexandraepan) [inline doc/daal4py/index.rst:149]:
```suggestion
See  
```

File: doc/daal4py/index.rst (hunk 1 above)
Comment (alexandraepan) [inline doc/daal4py/index.rst:193]:
```suggestion
    We recommend using 
```

File: doc/daal4py/model-builders.rst [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+.. * limitations under the License.
+.. *******************************************************************************/
+
+.. _model-builders:
+
+###############################################
+Model Builders for Gradinet Boosting Framewotrks
+###############################################
+Introduction
+---------
```
Comment (alexandraepan) [inline doc/daal4py/model-builders.rst:23]:
```suggestion
------------------
```

File: doc/daal4py/model-builders.rst [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  d4p_model = d4p.get_gbt_model_from_lightgbm(lgb_model)
+
+CatBoost::
+
+  import daal4py as d4p
+  d4p_model = d4p.get_gbt_model_from_catboost(cb_model)
+
+It's requered to convert model only once and then it will be used for inference
+
+Classification and Regression inference
```
Comment (alexandraepan) [inline doc/daal4py/model-builders.rst:53]:
```suggestion
Classification and Regression Inference
```

File: doc/daal4py/model-builders.rst [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+---------
+GBT implementaiton in daal4py assumes separate APIs for classification and regression.
+Corresponding API should be explicitly specified and should match corresponding problem
```
Comment (alexandraepan) [inline doc/daal4py/model-builders.rst:56]:
```suggestion
Specify the corresponding API and match the corresponding problem 
```

File: doc/daal4py/model-builders.rst [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+        nClasses=params['classes_count'],
+        resultsToEvaluate="computeClassLabels",
+        fptype='float'
+    )
+
+Regression::
+    d4p_reg_algo = d4p.gbt_regression_prediction()
+
+As a next step d4py algorithm object need compute method to be called. 
+Both data and previously converted model should be passed with results of predection
```
Comment (alexandraepan) [inline doc/daal4py/model-builders.rst:71]:
```suggestion
Both the data and the previously converted model should be passed with the results of the prediction 
```

File: doc/daal4py/model-builders.rst [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+        resultsToEvaluate="computeClassLabels",
+        fptype='float'
+    )
+
+Regression::
+    d4p_reg_algo = d4p.gbt_regression_prediction()
+
+As a next step d4py algorithm object need compute method to be called. 
+Both data and previously converted model should be passed with results of predection 
+avaialbe within .prediction parameter.
```
Comment (alexandraepan) [inline doc/daal4py/model-builders.rst:72]:
```suggestion
available within the  ``.prediction`` parameter.
```

File: doc/daal4py/model-builders.rst [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+    d4p_predictions = d4p_reg_algo.compute(X_test, d4p_model).prediction
+
+As alternative here is one line variant of same code::
+    d4p_prediction = d4p.gbt_regression_prediction().compute(X_test, d4p_model).prediction
+
+
+Limitations
+---------------------------------
+Missing Values (NaN)
+Note that there is temporary limitation on the use of missing values
```
Comment (alexandraepan) [inline doc/daal4py/model-builders.rst:85]: .. note:: There is a temporary limitation on the usage of the missing values

**Review by razdoburdin — APPROVED** [review]
Looks good for me

File: doc/daal4py/contents.rst [hunk 14]
```diff
@@ -25,8 +25,9 @@ Contents
 
    About daal4py <index>
    Data <data>
+   Model Builders <model-builders>
    Supported Algorithms <algorithms>
-   Distribution <scaling>
-   Streaming <streaming>
+   Distributed Compute <scaling>
+   Streaming Mode <streaming>
```
[the remark this answers is not in the document]
Comment (napetrov) [reply doc/daal4py/contents.rst]: Distributed Mode

**Pull request merged**